### ISSUE TITLE
Client caching for static content is broken when Preview is enabled

### ISSUE THREAD

**Issue by alanta** [Kentico/Mvc#19]
As soon as I enable the Preview module in an MVC site, client cache settings are completely ignored.

To reproduce:
* Setup an MVC-only Kentico site using the Kentico Nuget packages
* Setup application initialization as done in sample apps like Dancing Goat (App_Start\Application.cs)
* Add an entry to the routing table to ignore static content
* Now request a page and observe the headers on static content like /Content/Styles/Site.css
```
Cache-Control: private
```
* setup client caching in web.config:
```xml
<staticContent>
       <clientCache cacheControlCustom="public" cacheControlMaxAge="1.00:00:00" cacheControlMode="UseMaxAge"/>
</staticContent>
```
* The response headers should now include this:
```
Cache-Control: public,max-age=86400
```
But it doesn't.
* Now disable the Preview module in App_Start\Application.cs

After that, caching works as expected for static content.

I've traced this back to line 59 in PreviewFeatureModule.cs.
```csharp
context.Response.Cache.AddValidationCallback(ValidateCacheItem, null);
```
Commenting that line solves the issue. I'm not sure why that line is even there.
For static content, the event handler being registered is never hit.

**Comment by rpribyl** [comment]
I can confirm that the _cache-control_ header is being modified due to the added validation callback.
Validation callback registration ensures that pages displayed in the preview mode will not use their cached version if the output cache is enabled.

However there is an undocumented drawback in using the .NET `AddValidationCallback()` method.
When `AddValidationCallback` method is used, it causes that _cache_ headers will not be automatically populated.
See: [.NET HttpCachePolicy.cs](https://referencesource.microsoft.com/#System.Web/HttpCachePolicy.cs,562239178b9118a7)
Note the method `Dirtied()`

This drawback together with `RunAllManagedModulesForAllRequests="true"` and together with the fact that the callback registration is being done on `BeginRequest` causes that static content does not reflect the _web.config_ settings

The callback registration is needed by the preview feature BUT it should be moved from the `BeginRequest` further to the MVC pipeline.

**Comment by rpribyl** [comment]
Issue has been fixed here: [622e5e5](https://github.com/Kentico/Mvc/commit/622e5e5ecf4e6fe0d36c6dc3476b627a8ecd012a)
